lib/src/sock_utils/SockServer.cpp

```diff
@@ -137,18 +123,19 @@ sock_server_t* sock_server_init(int type, int port) {
         struct sockaddr_un addr_un;
         memset(&addr_un, 0, sizeof(addr_un));
         addr_un.sun_family = AF_UNIX;
-        strncpy(&addr_un.sun_path[0], SocketPath.c_str(), strlen(SocketPath.c_str()));
+        size_t length = (strlen(SocketPath.c_str()) < sizeof(addr_un.sun_path)) ?
+                        strlen(SocketPath.c_str()) : sizeof(addr_un.sun_path);
+        strncpy(&addr_un.sun_path[0], SocketPath.c_str(), length);
 
         int ret = 0;
         ALOGI(" %s sensor socket server file is %s", __FUNCTION__, SocketPath.c_str());
-        if ((access(SocketPath.c_str(), F_OK)) != -1) {
-            ret = unlink(SocketPath.c_str());
-            if (ret < 0) {
-                ALOGE(" %s Failed to unlink %s address %d, %s", __FUNCTION__,
-                                    SocketPath.c_str(), ret, strerror(errno));
-                close(socketfd);
-                return NULL;
-            }
+
+        ret = unlink(SocketPath.c_str());
```

Comment:
for the 1st time this will fail as the socket path might not exist. isnt it better to just log and proceed on unlink failure?